GitUI/Editor/FileViewerInternal.cs

```diff
@@ -131,19 +226,21 @@ public void SetText(string text, Action openWithDifftool, bool isDiff = false)
                 _lineNumbersControl.DisplayLineNumFor(text);
             }
 
+            int scrollPos = VScrollPosition;
             TextEditor.Text = text;
 
+            _currentViewPositionCache.Restore(isDiff);
+
+            // Reset the ScrollPos to stop annoying flicker when changing number of visible lines
+            VScrollPosition = scrollPos;
+
             // important to set after the text was changed
             // otherwise the may be rendering artifacts as noted in #5568
             TextEditor.ShowLineNumbers = ShowLineNumbers ?? !isDiff;
             if (ShowLineNumbers.HasValue && !ShowLineNumbers.Value)
             {
                 Padding = new Padding(DpiUtil.Scale(5), Padding.Top, Padding.Right, Padding.Bottom);
             }
-
-            TextEditor.Refresh();
-
-            _currentViewPositionCache.Restore(isDiff);
```

Comment:
We have to be really careful moving these calls around, we had issues such as #5568

Reply:
I have tested the swapping of files and can confirm it works as expected.
The TextEditor.Refesh is redundant but can be left in for safety.

_currentViewPositionCache.Restore(isDiff); 
must be done before:
TextEditor.ShowLineNumbers = ShowLineNumbers ?? !isDiff;
to prevent the flicker of the screen when increasing or decreasing the number of visible lines.